src/sagas/background-refresh.js

```diff
@@ -312,13 +332,49 @@ function* scheduleResumingNotifications ({ payload: { delayInSeconds } }) {
 function* logoutAndCancelScheduler () {
   yield put(Actions.setCurrentPage({ type: C.NO_REFRESH_TYPE }))
   yield put(Actions.stopSchedulerFixedDelay())
+  yield put(Actions.stopSchedulerForResumingNotifications())
+}
+
+// TODO: Remove after an upgrade to a redux-saga version that includes this effect
+// Adapted from https://redux-saga.js.org/docs/api#debouncems-pattern-saga-args
+function* debounce(interval, pattern, saga) {
+  while (true) {
+    let action = yield take(pattern)
+    console.info(`__ debounce take("${pattern}") 1 action:`, action)
+
+    while (true) {
+      const { debounced, latestAction } = yield race({
+        debounced: delay(interval),
+        latestAction: take(pattern),
+      })
+
+      console.info(`__ debounce take("${pattern}") 2: debounced:`, debounced, 'action:', action)
+      if (debounced) {
+        yield fork(saga, action)
+        break
+      }
+
+      action = latestAction
+    }
+  }
 }
 
 export default [
-  takeEvery(C.START_SCHEDULER_FIXED_DELAY, startSchedulerWithFixedDelay),
-  throttle(5000, C.MANUAL_REFRESH, refreshManually),
-  throttle(5000, C.REFRESH_DATA, refreshData),
+  debounce(5000, C.MANUAL_REFRESH, refreshManually),
+
+  /*
+   * Note: If a user goes crazy swapping between pages very quickly, a lot of `CHANGE_PAGE`
+   *       actions will be fired.  Since changePage() calls `START_SCHEDULER_FIXED_DELAY`
+   *       to refresh the new page data each time, it is possible to get to a point where
+   *       multiple page refresh sagas are running at the same time.  That can cause
+   *       problems and slow down the app.  To help prevent this without rearchitecting
+   *       the scheduler to support canceling running fetches, the action to start the
+   *       scheduler is debounced.  Now quick page changes that happen within 0.5 seconds
+   *       of each other won't generate excess refresh fetches.
+   */
   takeLatest(C.CHANGE_PAGE, changePage),
-  takeEvery(C.LOGOUT, logoutAndCancelScheduler),
+  debounce(500, C.START_SCHEDULER_FIXED_DELAY, startSchedulerWithFixedDelay),
```

Comment:
This (and above) `debounce()` solves the problem at the cost of slowing down every refresh. I think that the price is too high - I would give a try with replacing `debounce` with `throttle` or alternatively just ignore the problem and go back to original code here (assuming that we provide a fix in the near future).

Reply:
`throttle()` for the manual refresh makes sense ... work on the first one and ignore subsequent for the next 5 seconds.

`debounce()` still makes sense the start scheduler to make sure proper semantics are followed (i.e. only start the most recent one received in the debouce interval).  I'll turn the interval down to 1/4 second to make the delay less noticeable.